zerver/lib/message.py

```diff
@@ -870,7 +882,7 @@ def do_render_markdown(
         mention_data=mention_data,
         email_gateway=email_gateway,
     )
-    return rendered_content
+    return rendered_content, rendered_message
```

Comment:
I think there's two sensible ways to implement this interface:
* Approach one is to subclass `Message` in the `RenderedMessage` class, set `rendered_message.rendered_content` and just return the `rendered_message` object.  Though before doing that, we'll need a preparatory refactor to save `message.rendered_content`.
* Approach two is to call this new class `MessageRenderingResult`, have it not subclass Message, pass it as an added parameter to `markdown_convert`, and have it include a `.rendered_content` property when it is returned.

I think approach two might be cleaner in the end, but in any case, I don't think we want to return a tuple in either case.

BTW I think strategically it's pretty reasonable to just do this change as a new PR on top of master, having identified what we need to do from running the Django stubs -- that way, we'll get CI to help us verify that it works in master.

Reply:
Yes, I would prefer approach two as well.

Reply:
Great, let's go with that!  I think done correctly, this will fix a lot more than just the 20 errors you'd observed fixed with the original version.